view/mongo/src/main/kotlin/io/holunda/polyflow/view/mongo/Converters.kt

```diff
@@ -80,6 +80,31 @@ fun DataEntryUpdatedEvent.toDocument(oldDocument: DataEntryDocument?) = if (oldD
   )
 }
 
+/**
+ * Anonymized event to document.
+ */
+fun DataEntryAnonymizedEvent.toDocument(oldDocument: DataEntryDocument): DataEntryDocument {
+  val authorizedUsers = AuthorizationChange.applyUserAuthorization(
+    oldDocument.getAuthorizedUsers(),
+    oldDocument.getAuthorizedUsers().map { AuthorizationChange.removeUser(it) })
+
+  val protocol = oldDocument.protocol.map { it.toProtocol() }.map {
+    it.copy(
+      username = if (it.username != null && !this.excludedUsernames.contains(it.username)) this.anonymizedUsername else it.username
+    )
+  }.addModification(
```

Comment:
Same as in JPA. First add the modification, then anonymize.

Reply:
See comment above. I can change it but this is correct in my opinion